x-pack/plugins/alerting/server/rules_client/methods/bulk_delete.ts

```diff
@@ -35,54 +37,32 @@ export const bulkDeleteRules = async (context: RulesClientContext, options: Bulk
     action: 'DELETE',
   });
 
-  const { rules, errors, accListSpecificForBulkOperation } = await retryIfBulkOperationConflicts({
-    action: 'DELETE',
-    logger: context.logger,
-    bulkOperation: (filterKueryNode: KueryNode | null) =>
-      bulkDeleteWithOCC(context, { filter: filterKueryNode }),
-    filter: kueryNodeFilterWithAuth,
-  });
+  const { rules, errors, accListSpecificForBulkOperation } = await withSpan(
+    { name: 'retryIfBulkOperationConflicts', type: 'rules' },
+    () =>
+      retryIfBulkOperationConflicts({
+        action: 'DELETE',
+        logger: context.logger,
+        bulkOperation: (filterKueryNode: KueryNode | null) =>
+          bulkDeleteWithOCC(context, { filter: filterKueryNode }),
+        filter: kueryNodeFilterWithAuth,
+      })
+  );
 
   const [apiKeysToInvalidate, taskIdsToDelete] = accListSpecificForBulkOperation;
 
-  const taskIdsFailedToBeDeleted: string[] = [];
-  const taskIdsSuccessfullyDeleted: string[] = [];
-  if (taskIdsToDelete.length > 0) {
-    try {
-      const resultFromDeletingTasks = await context.taskManager.bulkRemoveIfExist(taskIdsToDelete);
-      resultFromDeletingTasks?.statuses.forEach((status) => {
-        if (status.success) {
-          taskIdsSuccessfullyDeleted.push(status.id);
-        } else {
-          taskIdsFailedToBeDeleted.push(status.id);
-        }
-      });
-      if (taskIdsSuccessfullyDeleted.length) {
-        context.logger.debug(
-          `Successfully deleted schedules for underlying tasks: ${taskIdsSuccessfullyDeleted.join(
-            ', '
-          )}`
-        );
-      }
-      if (taskIdsFailedToBeDeleted.length) {
-        context.logger.error(
-          `Failure to delete schedules for underlying tasks: ${taskIdsFailedToBeDeleted.join(', ')}`
-        );
-      }
-    } catch (error) {
-      context.logger.error(
-        `Failure to delete schedules for underlying tasks: ${taskIdsToDelete.join(
-          ', '
-        )}. TaskManager bulkRemoveIfExist failed with Error: ${error.message}`
-      );
-    }
-  }
-
-  await bulkMarkApiKeysForInvalidation(
-    { apiKeys: apiKeysToInvalidate },
-    context.logger,
-    context.unsecuredSavedObjectsClient
-  );
+  const [result] = await Promise.allSettled([
```

Comment:
what about the bulkMarkApiKeysForInvalidation result? what if one of these call fails? couldn't there be data inconsistency?

Reply:
No. it's ok. Both functions in Promise.allSettled do not throw the error. So we are ok that they can fail. Bulk delete method does not react even if bulkMarkApiKeysForInvalidation fails. We are count it's as ok.
To avoid error I even used allSettled, which does nor fail, but just return `rejected' as a status.